docs/FAQ.md

```diff
@@ -1,15 +1,159 @@
+carry out TXT on Intel, or SKINIT is used on AMD. TrenchBoot instead utilizes
+its own system, which is a custom version of GRUB that has commands built in to
+carry out a TrenchBoot "Secure Launch". This customized GRUB introduces the
+`slaunch` and `slaunch_module` commands. A working example of the TrenchBoot
+GRUB can be found at
+[https://github.com/3mdeb/meta-trenchboot](
+    https://github.com/3mdeb/meta-trenchboot).
+The custom GRUB works with Linux and Multiboot2 boot protocols for both Intel
+and AMD devices, but only in case of legacy boot, without support for UEFI
+(although it's available on a separate branch).
+
+### Intermediate Phase - Linux Kernel / U-ROOT (Initramfs)
+
+The intermediate phase processes information collected in the bootstrap phase
+and converts it into normalized data. This data is eventually used to determine
+whether to boot into the target OS. The kernel and u-root initramfs
+together make up Trenchboot's "Security Engine", which is an intermediate
+mini-OS that serves to process data gathered by that bootstrap phase. During the
+intermediate phase, the kernel and initramfs work together to measure block
+devices, individual files, SMBUS/DMI information and provides unseal based,
+external device and network based attestation. Trenchboot calls this
+functionality secure launch (aka slaunch). The table in section 4 lists the
+specific repo and branch to use when building the kernel or u-root.
+
+## 3. How to use Trenchboot? (practical usage)
+
+### Trenchboot GRUB TXT Config with Serial Communication
+
+*Note: ACM modules can be downloaded from [Intel's site][acm-download] (look for
+"Production SINIT ACM Download").*
+
+[acm-download]: https://www.intel.com/content/www/us/en/developer/articles/tool/intel-trusted-execution-technology.html
+
+```text
+menuentry 'TrenchBoot' {
+        savedefault
+        load_video
+        set gfxpayload=keep
+        insmod gzio
+        insmod part_msdos
+        insmod ext2
+        set root='hd0,msdos1'
+        multiboot2 /boot/slboot.gz loglvl=all serial=115200,8n1,0x3f8 logging=serial,memory
+        module2 /boot/vmlinuz-trenchboot root=UUID=<your block uuid> ro intel_iommu=on loglevel=7 console=ttyS0,115200 nokaslr
+        slaunch
+        slaunch_module /boot/<your acm>.bin
+}
+```
+
+### Testing the System
+
+QEMU has been tested to boot fully into u-root while simulating the Trenchboot
+kernel.
+QEMU For external u-root
+
+```bash
+qemu-systemx86_64 -kernel <kernel repo>/arch/x86/boot/bzImage -initrd /tmp/initramfs.linux_amd64.cpio -append console=ttyS0,38400 -serial file:serial.out
+```
+
+QEMU with u-root built-in
+
+```bash
+qemu-systemx86_64 -kernel <kernel location>
+```
+
+## 4. Why does TrenchBoot use an intermediate launcher?
 
 For Linux systems doing both verified(secure) and measured boot, there is an
 intermediary that handles the security enforcement. For verified boot it is the
```

Comment:
"For verified boot, the intermediary is the UEFI shim loader and for measured boot, it is tboot."

Edited-by: Amanda Ferguson [af.jenks@gmail.com](mailto:af.jenks@gmail.com)